### PR TITLE
mon: drop ignored mdsbeacon

### PR DESCRIPTION
**Pull request by tchaikov**

### CODE REVIEW

**Comment by tchaikov** [comment]
@jecluis  could you take a look at this ? thanks. i also scheduled a rados qa run for this. will post the result once it completes.

File: src/mon/MDSMonitor.cc [hunk 1]
```diff
@@ -248,16 +248,17 @@ bool MDSMonitor::preprocess_beacon(MMDSBeacon *m)
 
   // check privileges, ignore if fails
   MonSession *session = m->get_session();
-  if (!session)
-    goto out;
+  assert(session);
   if (!session->is_capable("mds", MON_CAP_X)) {
     dout(0) << "preprocess_beacon got MMDSBeacon from entity with insufficient privileges "
 	    << session->caps << dendl;
+    mon->no_reply(m);
```
Comment (gregsfortytwo) [inline src/mon/MDSMonitor.cc]: Are there other places that "goto out" and don't invoke no_reply()? Can we just add a label/cleanup thing that they all use?

Comment (tchaikov) [inline src/mon/MDSMonitor.cc]: > Are there other places that "goto out" and don't invoke no_reply()?
no

> Can we just add a label/cleanup thing that they all use?

yes, we can. and i pondered this option also, somehow i traded this label and a `m->put()` with these `mon->no_reply(m)` calls.  yes, this way is more consistent with the `ignore` handling. will do.

**Comment by tchaikov** [comment]
@gregsfortytwo thanks for your review! updated accordingly.

**Comment by gregsfortytwo** [comment]
Reviewed-by:

I think the session checks were from before we mocked up fake sessions for the forwarded messages? @jecluis should check this over too though.

**Comment by jecluis** [comment]
I think this is correct.

**Comment by tchaikov** [comment]
i ran a 1/18 rados suite with this pull request at http://pulpito.ceph.redhat.com/kchai-2015-06-03_05:51:30-rados-wip-kefu-testing---basic-magna/ and the re-ran the failed ones at http://pulpito.ceph.redhat.com/kchai-2015-06-04_10:57:04-rados-wip-kefu-testing---basic-magna and http://pulpito.ceph.redhat.com/kchai-2015-06-04_10:56:19-rados-wip-kefu-testing---basic-magna, and the two failed tests passed.

**Comment by gregsfortytwo** [comment]
http://pulpito.ceph.com/ubuntu-2015-06-09_15:15:42-fs-greg-fs-testing---basic-multi/

**Pull request merged**